services/searcher.ts

```diff
@@ -458,6 +459,60 @@ class Searcher {
     };
   }
 
+  filterSection(
+    curSection: Section,
+    honorsFilter: boolean,
+    campusFilter: string
+  ): boolean {
+    if (honorsFilter && !curSection.honors) {
+      return false;
+    }
+    if (campusFilter !== "" && curSection.campus !== campusFilter) {
+      return false;
+    }
+
+    return true;
+  }
+
+  filterResults(filters, results): SearchResult[] {
+    // Do extra filtering if Honors or Campus Filters are being applied since those filter sections,
+    // which ElasticSearch doesn't actually filter out.
+
+    const validFilters = this.validateFilters(filters);
+
+    const honorsFilter: boolean = validFilters["honors"] ? true : false;
+
+    const campusFilter: string = "" + (validFilters["campus"] ?? "");
```

Comment:
```suggestion
    const campusFilter = validFilters["campus"] ?? "";
```

Same comment as above - we don't want to use the `"" + ...` pattern to forcibly convert `validFilters["campus"]` to a string, since that can mask errors. Also, the explicit type isn't necessary here - TS is good at figuring out basic types like this (you can leave it, ofc, just figured it's worth pointing out)